### PR TITLE
add support for `/permissive-`

### PR DESCRIPTION
**Pull request by ubsan**
We do this with `cpp_std=vc++NM` - `vc++` does not add `/permissive-`, while `cpp_std=c++NM` adds `/permissive-`.

### CODE REVIEW

**Comment by jpakkane** [comment]
Github seems to be broken and no new CI runs are being started. This is probably due to the service outage. Hopefully it will get fixed soon.

File: mesonbuild/compilers/cpp.py [hunk 1]
```diff
@@ -338,10 +338,20 @@ def get_option_compile_args(self, options):
...
-        std = options['cpp_std']
-        if std.value == 'none':
+        vc_version_map = {
+            'none': (True, None),
+            'vc++11': (True, 11),
+            'vc++14': (True, 14),
+            'vc++17': (True, 17),
+            'c++11': (False, 11),
+            'c++14': (False, 14),
+            'c++17': (False, 17) }
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/cpp.py]: __[Flake8]__

[E202] whitespace before '}'

**Comment by ubsan** [comment]
@jpakkane is there a good way to skip the test before it checks if the `cpp_std` is accurate?

**Comment by jpakkane** [comment]
On your own machine you can do the standard Python unittest thing of `./run_unittests WindowsTests.test_msvc_cpp17` but on the CI not really. Apart from deleting test code or rearranging the order in which they are run.

**Comment by ubsan** [comment]
fair; it just seems unfortunate that the CI is failing when it's doing the correct thing.

**Comment by ubsan** [comment]
The tests seem to be failing on Boost `typeof`, which is a library which seems to have not been updated in 13 years.

**Pull request merged**